backend/core/api_serializers/author_api_serializer.py

```diff
@@ -0,0 +1,71 @@
+# Third-party libraries
+from rest_framework import serializers
+
+# Local libraries
+from ..models import *
+from ..serializers.authorserializer import AuthorSerializer
+
+author_serializer = AuthorSerializer()
+
+class AuthorAPISerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Author
+        fields = "__all__"
+
+    def get_author_data(self, author):
+        author_data = {
+            "type": "author",
+            "id": author.url,
+            "url": author.url,
+            "host": author.host,
+            "displayName": f"{author.first_name} {author.last_name}",
+            "username": author.username,
+            "email": author.email,
+            "profileImage": author.avatar,
+            }
+        return author_data
+    
+    def get_single_author(self, author_id):
+        author_data = {}
+        try:
+            author = author_serializer.get_author_by_id(author_id)
+            author_data = self.get_author_data(author)
+        except ValueError:
+            return None
+        return author_data
+    
+    def get_all_authors(self):
+
+        authors = Author.objects.all()
+        
+        result_dict = {}
+        result_dict["type"] = "authors"
+
+        authors_list = []
+
+        for author in authors:
+            curr_author_data = self.get_author_data(author)
+            authors_list.append(curr_author_data)
+
+        result_dict["items"] = authors_list
+
+        return result_dict
+
+    def create_or_update_author(self, author_id, request_data):
+        author = None
```

Comment:
Why do we need a separate method for create_or_update_author?

Reply:
Like could we rename the method name

Reply:
Yes we should rename this one to just update.